core/src/main/java/io/questdb/network/DefaultIODispatcherConfiguration.java

```diff
@@ -103,4 +103,10 @@ public int getTestConnectionBufferSize() {
     public long getTimeout() {
         return 5 * 60 * 1000L;
     }
+
+    @Override
+    public long getHeartbeatInterval() {
+        // don't send heartbeat messages by default
+        return -1L;
```

Comment:
why cant we return correct value right away ?

Reply:
to make it consistent with a disconnect on the idle logic. a negative value effectively disables the heartbets for all contexts except ILP, for which it only makes sense for now.

Reply:
what if it returned Long.MIN_VALUE ? this would remove checks we do and simplify the flow ?

Reply:
Long.MIN_VALUE will work like any other negative value. but we should keep the check because the intention is not to adjust the configuration but to disable it completely

Reply:
ok,  cool, lets leave it